### PR TITLE
new(libscap): add parsers for gVisor seccheck messages

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area driver-kmod

> /area driver-ebpf

/area libscap

> /area libsinsp

/area tests

> /area proposals

**What this PR does / why we need it**:

This is the first of a series of PRs to add support to Falco to process security/syscall events coming from [gVisor](https://gvisor.dev). 
gVisor is able to send syscall events to a remote checker via protobuf (see [this commit](https://github.com/google/gvisor/commit/2a238b23e7c2440110f436619b960ce77cefa04e) where the feature was introduced). This PR adds the necessary library functions in libscap to:
- parse the protobuf messages
- convert a number of syscall events to scap format, suitable for Falco
- add unit tests for this functionality

This work (including later PRs) has been a team effort with @loresuso with a lot of great help from @fvoznika from the gVisor team, and myself.

In order to keep this reviewable by maintainers this PR does not include support for connecting to gVisor and streaming events, which will be introduced with a subsequent PR.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
new(libscap): add parsers for gVisor seccheck messages
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Yay!!!
/ok-to-test

File: userspace/libscap/engine/gvisor/gvisor.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+#pragma pack(push, 1)
+struct header
+{
+	uint16_t header_size;
+	uint32_t dropped_count;
+};
+#pragma pack(pop)
+
+struct parse_result {
```
Comment (FedeDP) [inline userspace/libscap/engine/gvisor/gvisor.h:43]: Mmmh `parse_result` sounds part of `parsers` namespace; wdyt?

Comment (LucaGuerra) [reply userspace/libscap/engine/gvisor/gvisor.h:46]: Makes sense! Thanks!

**Review by FedeDP — APPROVED** [review]
This looks clean to me.
/approve

**Comment by FedeDP** [comment]
I see, thanks! What about inlining them?

**Review by FedeDP — APPROVED** [review]
/approve

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**